server/src/main/java/org/opensearch/transport/TransportSettings.java

```diff
@@ -290,7 +290,7 @@ public final class TransportSettings {
     public static final Setting<Integer> CONNECTIONS_PER_NODE_STATE = intSetting(
         "transport.connections_per_node.state",
         1,
-        1,
+        0,
```

Comment:
Can we have some validations on minimum connection by checking for remote store configurations enabledn on the node

Reply:
@Bukhtawar This is the validation for minimum connections only right ? This change is for minimum required connections validation only.
I.e, with this change, now someone can set minimum connection to `0` instead `1` earlier.

Reply:
I think we should have validation that a small cluster (2 Cluster Manager / 2 Data Nodes?) with these settings all dialed to zero can come up and handle requests that would depend on these different kind of connections.  Verifying that all operations succeeded and the connections spun up / down as expected.

This would ensure predictability in this space - and make sure if there is a regression and these low number of connections cause functionality breaks those regressions would be caught.

Reply:
+1 Lets add an integ test that uses the minimal values and verify if node bootstrap doesn't see any regression

Reply:
Actully I am closing this PR @peternied @Bukhtawar ,
we have already such checks in place to make number of connections to `0` irrespective of what is being passed from config. check out https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/transport/ConnectionProfile.java#L108.